Spigot-Server-Patches/0234-Anti-Xray.patch

```diff
@@ -0,0 +1,1426 @@
+From ab5f0dfccde97e80a7f970afb62245806dab793a Mon Sep 17 00:00:00 2001
+From: stonar96 <minecraft.stonar96@gmail.com>
+Date: Fri, 18 Aug 2017 13:27:37 +0200
+Subject: [PATCH] Anti-Xray
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 54d081fd..5c7106e4 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -1,5 +1,6 @@
+ package com.destroystokyo.paper;
+ 
++import java.util.Arrays;
+ import java.util.List;
+ 
+ import net.minecraft.server.MinecraftServer;
+@@ -428,4 +429,22 @@ public class PaperWorldConfig {
+         disableCreeperLingeringEffect = getBoolean("disable-creeper-lingering-effect", false);
+         log("Creeper lingering effect: " + disableCreeperLingeringEffect);
+     }
++
++    public boolean antiXray;
++    public boolean asynchronous;
++    public int engineMode;
++    public int chunkEdgeMode;
++    public int maxChunkSectionIndex;
++    public List<Object> hiddenBlocks;
++    public List<Object> replacementBlocks;
++    private void antiXray() {
++        antiXray = getBoolean("anti-xray.enabled", false);
++        asynchronous = true;
++        engineMode = getInt("anti-xray.engine-mode", 2);
++        chunkEdgeMode = getInt("anti-xray.chunk-edge-mode", 1);
++        maxChunkSectionIndex = getInt("anti-xray.max-chunk-section-index", 3);
++        hiddenBlocks = getList("anti-xray.hidden-blocks", Arrays.asList((Object) "gold_ore", "iron_ore", "coal_ore", "lapis_ore", "mossy_cobblestone", "obsidian", "chest", "diamond_ore", "redstone_ore", "lit_redstone_ore", "clay", "emerald_ore", "ender_chest"));
++        replacementBlocks = getList("anti-xray.replacement-blocks", Arrays.asList((Object) "stone", "planks"));
++        log("Anti-Xray: " + (antiXray ? "enabled" : "disabled") + " / Engine Mode: " + engineMode + (engineMode == 1 ? " (hide ores)" : " (obfuscate)") + " / Chunk Edge Mode: " + chunkEdgeMode + (chunkEdgeMode == 1 ? " (default)" : chunkEdgeMode == 2 ? " (wait until nearby chunks are loaded)" : " (load nearby chunks)") + " / Obfuscating up to " + ((maxChunkSectionIndex + 1) * 16) + " blocks");
```

Comment:
Enum should be used here, and on the field, too.
Additionally, the enum should be adjusted to contain a description:
```java
HIDE(1, "hide ores"),
OBFUSCATE(2, "obfuscate");
```
and use that here.